Fix CSC, DT, and RPC digi validation configuration for premixing

makortel opened the pull request:
This PR suggests fixing the CSC, DT, and RPC digi validation configurations for premixing as noted in HN message 
https://hypernews.cern.ch/HyperNews/CMS/get/muon-performance/982.html
as well as `GlobalDigisAnalyzer` configuration.

The fix requires updating the event content for the premixing stage2 DIGI output to contain the SimDigis from `mixData`. To reduce confusion I decided to drop the signal-only SimDigi+DigiSimLink collections.

Tested in CMSSW_10_2_X_2018-04-12-2300, expecting changes in CSC, DT, and RPC digi validation histograms a well as in "GlobalDigis" validation in premixing workflows, but nothing else.

@mdhildreth @kpedro88

makortel committed:
Fix CSC, DT, and RPC digi validation configurations for premixing

makortel committed:
Add correct muon SimDigis to event content for premixing, and drop signal-only ones

makortel committed:
Fix muon digi configuration for premixing in GlobalDigisAnalyzer as well

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22977/4362

cmsbuild commented:
A new Pull Request was created by  @makortel (Matti Kortelainen) for master.

It involves the following packages:

Configuration/EventContent
SimMuon/Configuration
Validation/GlobalDigis
Validation/MuonCSCDigis
Validation/MuonDTDigis
Validation/MuonRPCDigis


@civanch, @vazzolini, @kmaeshima, @mdhildreth, @dmitrijus, @cmsbuild, @franzoni, @jfernan2, @vanbesien, @fabiocos, @davidlange6 can you please review it and eventually sign? Thanks.
@battibass, @abbiendi, @jhgoh, @Martin-Grunewald, @calderona, @HuguesBrun, @drkovalskyi, @dildick, @trocino, @folguera this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

makortel commented:
@cmsbuild, please test workflow 250202.0,250202.17,250202.18,250402.0

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 0d62b2109c30c3fd221f0058f3053acf930f00df
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22977/27514/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22977/27514/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-22977/250202.18_TTbar_13UP18+TTbar_13UP18+DIGIPRMXUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-22977/250402.0_TTbar_13+FS_TTbar_13_PRMXUP15_PU25+HARVESTUP15FS+MINIAODMCUP15FS

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 29
 - DQMHistoTests: Total histograms compared: 2492536
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2492359
 - DQMHistoTests: Total skipped: 176
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 23 files compared)
 - Checked 119 log files, 9 edm output root files, 29 DQM output files

dmitrijus commented:
+1

civanch commented:
+1

civanch commented on Validation/GlobalDigis/python/globaldigis_analyze_cfi.py:
```diff
@@ -31,5 +31,10 @@
+    MuCSCStripSrc = "mixData:MuonCSCStripDigisDM",
+    MuCSCWireSrc = "mixData:MuonCSCWireDigisDM",
+    MuDTSrc = "mixData",
+    MuRPCSrc = "mixData",
```
@makortel , may be I am wrong but is it OK to have comma in line 39?

kpedro88 replied:
Indeed, I typically recommend using the trailing comma because it makes git diffs cleaner when new lines are added in a list.

fabiocos commented:
@makortel @dmitrijus @civanch are we sure that the dropped collections are really unused in every possible workflow?

makortel commented:
@fabiocos All event content adjustments are specific for premixing stage2. Dropped collections correspond to signal-only digis, and added signal+pileup digis. I'd argue that using signal-only digis would be a mistake and should be fixed.

fabiocos commented:
@makortel running 10824.0 after merging this PR into a test area, I see that in the output of the step2 (that is based on FEVTDEBUGHLT) the collections are still present:

edmEventSize -v step2.root | grep DigiSimLink | awk '{print $1}'
StripDigiSimLinkedmDetSetVector_simSiStripDigis__HLT.
PixelDigiSimLinkedmDetSetVector_simSiPixelDigis__HLT.
RPCDigiSimLinkedmDetSetVector_simMuonRPCDigis_RPCDigiSimLink_HLT.
DTLayerIdDTDigiSimLinkMuonDigiCollection_simMuonDTDigis__HLT.
StripDigiSimLinkedmDetSetVector_simMuonCSCDigis_MuonCSCStripDigiSimLinks_HLT.
StripDigiSimLinkedmDetSetVector_simMuonCSCDigis_MuonCSCWireDigiSimLinks_HLT.
StripDigiSimLinkedmDetSetVector_simMuonGEMDigis_GEM_HLT.
GEMDigiSimLinkedmDetSetVector_simMuonGEMDigis_GEM_HLT.

Probably because they anyway enter through the SimMuonFEVTDEBUG content. So your change looks to be just removing a redundant declaration.
And these collections are anyway used as an input by the SimMuon/MCTruth associator. So my conclusion is that this PR is not really modifying the event content, and if it was doing so it would break the associator. Do you agree?

makortel commented:
@fabiocos 10824.0 is a noPU workflow, while the event content changes in this PR (should) affect only premixing. So no changes in10824.0 is as expected. If you want to see the changes, please see e.g. 250202.18 `step2.root`.

fabiocos commented:
@makortel yes, I see the change in 250202.18. Ok, so I believe it should be ok

fabiocos commented:
+operations

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

fabiocos commented:
+1
